src/client/coap_client.erl

```diff
@@ -117,7 +124,8 @@ split_path([$/]) -> [];
 split_path([$/ | Path]) -> split_segments(Path, $/, []).
 
 split_query([]) -> [];
-split_query([$? | Path]) -> split_segments(Path, $&, []).
+split_query([$? | Path]) -> split_segments(Path, $&, []);
```

Comment:
the new uri parser do not return query string with leading `?`, we can delete this clause.